flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -1244,4 +1278,99 @@ private static List<RowData> fetchAllResults(ResultFetcher resultFetcher) {
         }
         return results;
     }
+
+    private JobExecutionResult executeJob(
+            String script,
+            Configuration executionConfig,
+            OperationExecutor operationExecutor,
+            OperationHandle operationHandle) {
+        if (isApplicationMode(operationExecutor.getSessionContext().getSessionConf())) {
```

Comment:
Let's check the target is not null and not local before execute job, otherwise throw exception. 

For MT,   I think we only support four deploy cluster modes: minicluster, standalone, yarn, k8s, so we only need to consider them, so the corresponding `execution.target`  we only need to consider `remote`, `yarn`, `k8s`.  For minicluster and standalone clusters, `execution.target` is the remote, yarn is `yarn-session` and `yarn-application`, k8s is `kubernetes-session` and `kubernetes-application`. 

When `execution.target` is local, I don't know what is it and its function, we don't need to support it.

Reply:
I spent some time researching the execution mode of local. For the local mode, the corresponding execution environment is minicluster. So I understand that we need to support the local mode? In this case, clusterId may also be null.

Reply:
I have tested locally. When I set the execution.mode of sql gateway to local, the job will start with minicluster.